<H1> Johdanto datatieteeseen - Harjoitustyö </H1>

<H2> Datan kerääminen </H2>

Ladataan Berliinin Airbnb-data Inside Airbnb -sivuilta ja tuodaan se csv-muodossa jupyteriin

Valitaan halututut sarakkeet ja luodaan uusi dataframe. Tämän jälkeen esitetään 100 riviä datasta

In [33]:
LISTINGSFILE = 'listings.csv'
cols = ['price',
        'accommodates',
        'bedrooms',
        'beds',
        'bathrooms',
        'room_type',
        'neighbourhood_group_cleansed',
        'number_of_reviews',
        'latitude',
        'longitude'
        ]
df = pd.read_csv(LISTINGSFILE, usecols=cols)
pd.set_option('display.max_rows', 100)
df

neighbourhood_group_cleansed   latitude  longitude        room_type  \
0                           Pankow  52.583306  13.396092     Private room   
1                           Pankow  52.576417  13.398582     Private room   
2                           Pankow  52.589218  13.399261     Private room   
3                           Pankow  52.582312  13.395584     Private room   
4                           Pankow  52.583749  13.393662     Private room   
5                           Pankow  52.579673  13.399317     Private room   
6                           Pankow  52.587612  13.400317     Private room   
7                           Pankow  52.596733  13.403350     Private room   
8                           Pankow  52.587892  13.394384     Private room   
9                           Pankow  52.578969  13.399527     Private room   
10                          Pankow  52.588876  13.395327     Private room   
11                          Pankow  52.583823  13.396386     Private room   
12                          Pankow  52.577430  13.400585     Private room   
13                          Pankow  52.592778  13.409398     Private room   
14                          Pankow  52.591486  13.410317     Private room   
15                          Pankow  52.574690  13.398817     Private room   
16                          Pankow  52.582888  13.396139  Entire home/apt   
17                          Pankow  52.587518  13.401295  Entire home/apt   
18                          Pankow  52.581503  13.397858  Entire home/apt   
19                          Pankow  52.579581  13.402656  Entire home/apt   
20                          Pankow  52.583045  13.396751  Entire home/apt   
21                          Pankow  52.577105  13.400070  Entire home/apt   
22                          Pankow  52.588019  13.401749  Entire home/apt   
23                          Pankow  52.579925  13.401222  Entire home/apt   
24                          Pankow  52.582314  13.407388  Entire home/apt   
25                          Pankow  52.594405  13.393063  Entire home/apt   
26                          Pankow  52.574868  13.398947  Entire home/apt   
27                          Pankow  52.587738  13.401057     Private room   
28                          Pankow  52.576065  13.399548  Entire home/apt   
29                          Pankow  52.598707  13.389155     Private room   
30                          Pankow  52.581875  13.395727     Private room   
31                          Pankow  52.586121  13.394648     Private room   
32                          Pankow  52.585326  13.411020  Entire home/apt   
33                          Pankow  52.583494  13.401539     Private room   
34                          Pankow  52.586284  13.408938  Entire home/apt   
35                          Pankow  52.577453  13.393781  Entire home/apt   
36                          Pankow  52.579533  13.399755     Private room   
37                          Pankow  52.584069  13.401456  Entire home/apt   
38                          Pankow  52.588421  13.401595  Entire home/apt   
39                          Pankow  52.576788  13.399374     Private room   
40                          Pankow  52.587502  13.392901  Entire home/apt   
41                          Pankow  52.576861  13.396443     Private room   
42                          Pankow  52.579452  13.396910     Private room   
43                          Pankow  52.588370  13.423595  Entire home/apt   
44                          Pankow  52.543697  13.424874     Private room   
45                          Pankow  52.544158  13.424006     Private room   
46                          Pankow  52.542453  13.424477     Private room   
47                          Pankow  52.545814  13.422472     Private room   
48                          Pankow  52.541615  13.423260     Private room   
49                          Pankow  52.539790  13.422996     Private room   
...                            ...        ...        ...              ...   
20

<H2> Datan jalostaminen </H2>

'nan'-arvojen etsiminen

In [34]:
pd.isnull(df).any()

neighbourhood_group_cleansed    False
latitude                        False
longitude                       False
room_type                       False
accommodates                    False
bathrooms                        True
bedrooms                         True
beds                             True
price                           False
number_of_reviews               False
dtype: bool

'nan'-arvojen poistaminen 

In [35]:
df = df.dropna(subset=['bathrooms', 'bedrooms', 'beds'])

Ylimääräisten sarakkeiden poistaminen 'price'-sarakkeesta ja arvojen muuttaminen numeerisiksi

In [43]:
df['price'] = df['price'].str.strip('$')
df['price'] = df['price'].str.replace(',','')
df['price'] = pd.to_numeric(df['price'])

AttributeError: Can only use .str accessor with string values, which use np.object_ dtype in pandas

Tulee virheilmoitus (toimii Spyderissa), tosin tehdyt muutokset 'price' -sarakkeeseen näkyvät dataframessa

In [44]:
df

neighbourhood_group_cleansed   latitude  longitude        room_type  \
0                           Pankow  52.583306  13.396092     Private room   
1                           Pankow  52.576417  13.398582     Private room   
2                           Pankow  52.589218  13.399261     Private room   
3                           Pankow  52.582312  13.395584     Private room   
4                           Pankow  52.583749  13.393662     Private room   
5                           Pankow  52.579673  13.399317     Private room   
6                           Pankow  52.587612  13.400317     Private room   
7                           Pankow  52.596733  13.403350     Private room   
8                           Pankow  52.587892  13.394384     Private room   
9                           Pankow  52.578969  13.399527     Private room   
10                          Pankow  52.588876  13.395327     Private room   
11                          Pankow  52.583823  13.396386     Private room   
12                          Pankow  52.577430  13.400585     Private room   
13                          Pankow  52.592778  13.409398     Private room   
14                          Pankow  52.591486  13.410317     Private room   
15                          Pankow  52.574690  13.398817     Private room   
16                          Pankow  52.582888  13.396139  Entire home/apt   
17                          Pankow  52.587518  13.401295  Entire home/apt   
18                          Pankow  52.581503  13.397858  Entire home/apt   
19                          Pankow  52.579581  13.402656  Entire home/apt   
20                          Pankow  52.583045  13.396751  Entire home/apt   
21                          Pankow  52.577105  13.400070  Entire home/apt   
22                          Pankow  52.588019  13.401749  Entire home/apt   
23                          Pankow  52.579925  13.401222  Entire home/apt   
24                          Pankow  52.582314  13.407388  Entire home/apt   
25                          Pankow  52.594405  13.393063  Entire home/apt   
26                          Pankow  52.574868  13.398947  Entire home/apt   
27                          Pankow  52.587738  13.401057     Private room   
28                          Pankow  52.576065  13.399548  Entire home/apt   
29                          Pankow  52.598707  13.389155     Private room   
30                          Pankow  52.581875  13.395727     Private room   
31                          Pankow  52.586121  13.394648     Private room   
32                          Pankow  52.585326  13.411020  Entire home/apt   
33                          Pankow  52.583494  13.401539     Private room   
34                          Pankow  52.586284  13.408938  Entire home/apt   
35                          Pankow  52.577453  13.393781  Entire home/apt   
36                          Pankow  52.579533  13.399755     Private room   
37                          Pankow  52.584069  13.401456  Entire home/apt   
38                          Pankow  52.588421  13.401595  Entire home/apt   
39                          Pankow  52.576788  13.399374     Private room   
40                          Pankow  52.587502  13.392901  Entire home/apt   
41                          Pankow  52.576861  13.396443     Private room   
42                          Pankow  52.579452  13.396910     Private room   
43                          Pankow  52.588370  13.423595  Entire home/apt   
44                          Pankow  52.543697  13.424874     Private room   
45                          Pankow  52.544158  13.424006     Private room   
46                          Pankow  52.542453  13.424477     Private room   
47                          Pankow  52.545814  13.422472     Private room   
48                          Pankow  52.541615  13.423260     Private room   
49                          Pankow  52.539790  13.422996     Private room   
...                            ...        ...        ...              ...   
20

Otetaan huomioon vain nollaa suuremmat arvot

In [45]:
df = df[df.bedrooms > 0]
df = df[df.price > 0]
df = df[df.bathrooms > 0]
df = df[df.number_of_reviews > 0]

Luodaan kategorisista attribuuteista ja hinnasta uusi dataframe helpottamaan visualisointeja

In [46]:
df_filter = df.filter(['price','room_type','neighbourhood_group_cleansed'], axis=1)

Kategoristen arvojen muuttaminen numeerisiksi ja niiden lisääminen dataframeen sekä alkuperäisten kategoristen attribuuttien poistaminen

In [47]:
roomtype_dummies = pd.get_dummies(df.room_type)
neighbourhood_dummies = pd.get_dummies(df.neighbourhood_group_cleansed)

data = pd.concat((df.drop(['room_type', 'neighbourhood_group_cleansed'], \
    axis=1), roomtype_dummies.astype(int), neighbourhood_dummies.astype(int)),axis=1)
allcols = data.columns

In [49]:
data


latitude  longitude  accommodates  bathrooms  bedrooms  beds  price  \
0      52.583306  13.396092             2        1.0       1.0   2.0   30.0   
1      52.576417  13.398582             2        1.5       1.0   1.0   17.0   
2      52.589218  13.399261             4        1.0       1.0   1.0   50.0   
3      52.582312  13.395584             2        1.0       1.0   1.0   28.0   
4      52.583749  13.393662             1        1.0       1.0   1.0   21.0   
5      52.579673  13.399317             2        1.0       1.0   1.0   38.0   
8      52.587892  13.394384             1        1.0       1.0   1.0   37.0   
10     52.588876  13.395327             2        1.0       1.0   1.0   20.0   
11     52.583823  13.396386             1        1.0       1.0   1.0   35.0   
12     52.577430  13.400585             1        1.0       1.0   1.0   30.0   
13     52.592778  13.409398             2        1.0       1.0   1.0   27.0   
14     52.591486  13.410317             1        1.0       1.0   1.0   20.0   
15     52.574690  13.398817             2        1.0       1.0   2.0   36.0   
17     52.587518  13.401295             4        1.0       2.0   2.0   45.0   
18     52.581503  13.397858             4        1.0       2.0   2.0   69.0   
19     52.579581  13.402656             6        1.5       2.0   4.0   80.0   
21     52.577105  13.400070             4        1.0       2.0   3.0   49.0   
24     52.582314  13.407388             2        1.0       1.0   2.0   39.0   
26     52.574868  13.398947             6        1.0       3.0   6.0  120.0   
29     52.598707  13.389155             2        1.0       1.0   1.0   35.0   
31     52.586121  13.394648             1        1.0       1.0   1.0   35.0   
32     52.585326  13.411020             3        1.0       1.0   2.0   30.0   
33     52.583494  13.401539             2        1.0       1.0   1.0   39.0   
35     52.577453  13.393781             2        1.0       1.0   1.0   45.0   
36     52.579533  13.399755             1        1.0       1.0   1.0   15.0   
37     52.584069  13.401456             4        2.0       2.0   3.0  250.0   
39     52.576788  13.399374             3        1.0       1.0   1.0   20.0   
40     52.587502  13.392901             4        1.0       2.0   4.0   69.0   
41     52.576861  13.396443             3        2.0       1.0   2.0   25.0   
42     52.579452  13.396910             1        1.0       1.0   1.0   28.0   
43     52.588370  13.423595             5        1.0       1.0   4.0   85.0   
44     52.543697  13.424874             2        1.0       1.0   1.0   35.0   
45     52.544158  13.424006             2        1.0       1.0   1.0   23.0   
46     52.542453  13.424477             2        1.0       1.0   1.0   38.0   
47     52.545814  13.422472             1        1.0       1.0   1.0   22.0   
48     52.541615  13.423260             2        1.0       1.0   1.0   50.0   
49     52.539790  13.422996             2        1.0       1.0   1.0   55.0   
50     52.545670  13.420669             2        1.0       1.0   1.0   80.0   
51     52.544012  13.422029             3        1.0       1.0   2.0   64.0   
52     52.544258  13.424352             4        1.0       1.0   2.0   48.0   
53     52.544033  13.422371             2        1.0       1.0   1.0   79.0   
55     52.543194  13.424700             2        1.0       1.0   1.0   80.0   
56     52.548290  13.416929             2        1.0       1.0   1.0   48.0   
57     52.546786  13.415311             3        1.0       1.0   2.0   85.0   
58     52.540664  13.416857             2        1.0       1.0   1.0   42.0   
59     52.543387  13.426026             4        1.0       2.0   2.0  120.0   
60     52.544432  13.422218             4        1.0       1.0   2.0   52.0   
62     52.542699  13.420814             2        1.0       1.0   1.0   35.0   
63     52.542940  13.424285             2        1.0       1.0   1.0  108.0   
64     52.544872  13.423356             6        1.0      

In [50]:
data.shape

(15275, 23)

Jalostetussa datassa on 15275 riviä ja 23 attribuuttia (alkuperäisessä datassa 20576 riviä)